lib/datadog/appsec/contrib/faraday/middleware.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module Datadog
+  module AppSec
+    module Contrib
+      module Faraday
+        # Middleware for Faraday
+        class Middleware < ::Faraday::Middleware
```

Comment:
I think the class name should describe "what" because now it's stating a fact which is not helpful if you want to understand what is going to happen.

And the most important, we have a check that `Middleware` is not already in the list of middlewares, hence we have non 0% chances that someone decides to call their middleware as `Middleware`

Reply:
but what is the chance that someone will name their middleware `Datadog::AppSec::Contrib::Faraday::Middleware`?

Reply:
yes, let's maybe call it `SSRFDetectionMiddleware`?

Reply:
changed naming and refactored patched a little bit in [Improve naming in AppSec faraday patcher](https://github.com/DataDog/dd-trace-rb/pull/4391/commits/22b711546a2b85aec9faf2c07e21881bee16b672)